autoload/tsuquyomi/es6import.vim

```diff
@@ -201,6 +204,29 @@ function! s:getShortenedPath(splitted_absolute_path, previous_shortened_path, mo
   return l:current_directory_name . l:path_separator . l:shortened_path
 endfunction
 
+function! s:getBaseUrlImportPath(module_absolute_path)
+  let [l:tsconfig, l:tsconfig_file_path] = s:getTsconfig(a:module_absolute_path)
+  let l:project_root_path = fnamemodify(l:tsconfig_file_path, ':h').'/'
+  " We assume that baseUrl is a path relative to tsconfig.json path.
+  let l:base_url_config = has_key(l:tsconfig.compilerOptions, 'baseUrl') ? l:tsconfig.compilerOptions.baseUrl : '.'
+  let l:base_url_path = simplify(l:project_root_path.l:base_url_config)
+
+  return s:removeTSExtensions(substitute(a:module_absolute_path, l:base_url_path, '', ''))
+endfunction
+
+let s:tsconfig = {}
+let s:tsconfig_file_path = ''
+
+function! s:getTsconfig(module_absolute_path)
+  if empty(s:tsconfig)
+    let l:project_info = tsuquyomi#tsClient#tsProjectInfo(a:module_absolute_path, 0)
+    let s:tsconfig_file_path = l:project_info.configFileName
+    let s:tsconfig = s:JSON.decode(join(readfile(s:tsconfig_file_path),''))
```

Comment:
`tsc --init`  command creates a JSON file such as following.  It's **not** valid as JSON file(comment or trailing comma), but TypeScript's parser can parse this JSON-like file.

```js
{
  "compilerOptions": {
    /* Basic Options */
    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', or 'ESNEXT'. */
    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
    // "lib": [],                             /* Specify library files to be included in the compilation:  */
    // "allowJs": true,                       /* Allow javascript files to be compiled. */
    // "checkJs": true,                       /* Report errors in .js files. */
    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
    // "outFile": "./",                       /* Concatenate and emit output to single file. */
    // "outDir": "./",                        /* Redirect output structure to the directory. */
    // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
    // "removeComments": true,                /* Do not emit comments to output. */
    // "noEmit": true,                        /* Do not emit outputs. */
    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
    // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
  }
}
```

So, this `s:JSON.decode` may throw some parse error. Should we catch it 🤔 ?

Reply:
Well, it doesn’t look like we have a choice. 🙃